addons/point_of_sale/static/src/js/Screens/ProductScreen/OrderWidget.js

```diff
@@ -52,6 +52,26 @@ odoo.define('point_of_sale.OrderWidget', function(require) {
                 isSingleItem: isAllowOnlyOneLot,
                 array: packLotLinesToEdit,
             });
+            if (!this.env.pos.picking_type.use_create_lots) {
+                const lots = payload['newArray'].map(item => item.text);
+                await this.rpc({
+                    model: 'stock.production.lot',
+                    method: 'check_lots_exist',
+                    args: [lots],
+                }).then(async (result) => {
+                    if (result.length > 0) {
+                        const { confirmed } = await this.showPopup('ConfirmPopup', {
+                            title: this.env._t('Serial Number(s) Doesn\'t Exist'),
+                            body: this.env._t('The following serial numbers do not exists: ') + result.join(', ') + '\n' + this.env._t('Do you want to delete them?'),
+                            confirmText: this.env._t('Delete'),
+                            cancelText: this.env._t('Cancel'),
+                        });
+                        if (confirmed) {
+                            payload.newArray = payload.newArray.filter(item => !result.includes(item.text));
+                        }
+                    }
+                });
+            }
```

Comment:
You can extract this as a method and put it in `PosModel`. Then access thru `this.env.pos`. Also, you don't need to `.then` the rpc call:
```js
                const result = await this.rpc({
                    model: 'stock.production.lot',
                    method: 'check_lots_exist',
                    args: [lots],
                });
                // do something with the result
```